### PR TITLE
DCD-1179: Use PSQL query to deduce DB version

### PR DESCRIPTION
**Pull request by errcode1202**
Vulnerability mitigation by removing the vulnerable dependencies 😛 

Reported vulnerability: https://snyk.io/vuln/SNYK-JAVA-IONETTY-564897

### CODE REVIEW

**Comment by nanux** [comment]
Doesn't look there is any change at all?

File: core/src/main/kotlin/com/atlassian/migration/datacenter/core/db/PostgresClientTooling.kt [hunk 1]
```diff
@@ -86,26 +83,40 @@ class PostgresClientTooling(private val applicationConfiguration: ApplicationCon
...
+        
+        try {
+            /*
+             * pg_dump availability is a hard requirements on this plugin. 
+             * If pg_dump is available we can assume psql will also be available.
+             * 
+             * The pg_dump utility is obtained by installing a postgres
+             * client library. These clients typically include psql.
+             */
+            val proc = ProcessBuilder("/usr/bin/psql",
```
Comment (nanux) [inline core/src/main/kotlin/com/atlassian/migration/datacenter/core/db/PostgresClientTooling.kt:102]: Is it always on this path? Is it possible that different distributions are not installing it to a different destination?

Comment (errcode1202) [reply core/src/main/kotlin/com/atlassian/migration/datacenter/core/db/PostgresClientTooling.kt:102]: The intention is to actually dynamically look this up. I wanted to get opinions first on this approach (to fix the vulnerability) before sinking more time into it.

Comment (errcode1202) [reply core/src/main/kotlin/com/atlassian/migration/datacenter/core/db/PostgresClientTooling.kt]: I've update the code to take this into account.

**Review by nanux — APPROVED** [review]
LGTM, pending build results.

**Review by tarka — APPROVED** [review]

**Pull request merged**